tiledb/sm/query/sparse_unordered_with_dups_reader.cc

```diff
@@ -362,31 +363,35 @@ Status SparseUnorderedWithDupsReader::add_result_tile(
     return Status::Ok();
   }
 
-  memory_used_for_coords_total_ += tiles_size;
-
-  result_tiles_.emplace_back(f, t, domain);
-
+  uint64_t tiles_size_qc = 0;
   if (!condition_.empty()) {
-    uint64_t tiles_size = 0;
     for (auto& name : condition_.field_names()) {
       // Calculate memory consumption for this tile.
       uint64_t tile_size = 0;
-      RETURN_NOT_OK(
-          get_attribute_tile_size(name, &result_tiles_.back(), &tile_size));
-      tiles_size += tile_size;
+      RETURN_NOT_OK(get_attribute_tile_size(name, f, t, &tile_size));
+      tiles_size_qc += tile_size;
     }
 
-    memory_used_qc_tiles_ += tiles_size;
-    if (memory_used_qc_tiles_ > memory_budget_qc_tiles) {
+    if (memory_used_qc_tiles_ + tiles_size_qc > memory_budget_qc_tiles) {
       *budget_exceeded = true;
+      return Status::Ok();
```

Comment:
Really wanting to return Status::Ok() here on budget exceeded rather than Status::<SomeSortOf>Error()? 
(similar below)

Reply:
Yes it is not an error condition and is fully handled by the caller.